botocore/args.py

```diff
@@ -810,6 +818,23 @@ def _compute_checksum_config(self, config_kwargs):
             valid_options=VALID_RESPONSE_CHECKSUM_VALIDATION_CONFIG,
         )
 
+    def _resolve_protocol(self, service_model):
+        # We need to ensure `protocols` exists in the metadata before attempting to
+        #  access it directly since referencing service_model.protocols directly will
+        #  raise an UndefinedModelAttributeError if protocols is not defined
```

Comment:
```suggestion
        # access it directly since referencing service_model.protocols directly will
        # raise an UndefinedModelAttributeError if protocols is not defined
```